src/neptune/metadata_containers/utils.py

```diff
@@ -29,6 +43,8 @@
 )
 from neptune.internal.utils.run_state import RunState
 
+DATE_FORMAT: str = "%Y-%m-%dT%H:%M:%S.%fZ"
```

Comment:
I don't really like it tbh, but not sure what to do. Do we have a way to predict the output format of date objects from backend?

Reply:
There is lib [dateutils.parser](https://dateutil.readthedocs.io/en/stable/parser.html) but it's external library and much slower so not worth the hassle imo.

I think its good idea to wrap datetime parsing in try catch and logging warning / error if format doesnt match datetime (ValueError is risen)

Reply:
Yes, that could work (I mean the try-catch). Was also thinking if this format is somehow set in the backend and perhaps it would be possible to expose it to the client with an endpoint for that purpose